### PR TITLE
add environment variables for provider options

### PR DESCRIPTION
**Pull request by rashmi43**

### CODE REVIEW

**Comment by siddarthpatel** [comment]
@rashmi43 Your PR is failing tests because of linting errors: 
http://prow.purple-chesterfield.com/log?job=check_ibm-commonui-operator&id=1237779396035612673

You can install the following (all necessary) linting tools and run a `make check` locally to see if they are passing before committing. 
https://github.com/IBM/ibm-commonui-operator/blob/master/docs/development.md

**Comment by rashmi43** [comment]
All tools except for mdl are running on my system.

File: deploy/olm-catalog/ibm-commonui-operator/1.1.0/ibm-commonui-operator.v1.1.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -348,17 +356,10 @@ spec:
       kind: CommonWebUI
       name: commonwebuis.operators.ibm.com
       version: v1alpha1
-      displayName: CommonWebUI service
```
Comment (ericabr) [inline deploy/olm-catalog/ibm-commonui-operator/1.1.0/ibm-commonui-operator.v1.1.0.clusterserviceversion.yaml]: add this back to the CSV

File: deploy/olm-catalog/ibm-commonui-operator/1.1.0/ibm-commonui-operator.v1.1.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -348,17 +356,10 @@ spec:
...
     - description: LegacyHeader is the Schema for the legacyHeader API
       kind: LegacyHeader
       name: legacyheaders.operators.ibm.com
       version: v1alpha1
-      displayName: LegacyHeader service
```
Comment (ericabr) [inline deploy/olm-catalog/ibm-commonui-operator/1.1.0/ibm-commonui-operator.v1.1.0.clusterserviceversion.yaml]: Add this all back to the CSV, it gets removed when the CSV is generated

File: deploy/olm-catalog/ibm-commonui-operator/1.1.0/ibm-commonui-operator.v1.1.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -328,7 +332,11 @@ metadata:
               "cloudPakVersion": "3.2.4",
               "clusterName": "mycluster",
               "defaultAdminUser": "admin",
+              "defaultAuth": "",
```
Comment (ericabr) [inline deploy/olm-catalog/ibm-commonui-operator/1.1.0/ibm-commonui-operator.v1.1.0.clusterserviceversion.yaml]: this is for legacyheader, needs to be removed

File: deploy/olm-catalog/ibm-commonui-operator/1.1.0/operators.ibm.com_legacyheaders_crd.yaml [hunk 4]
```diff
@@ -77,6 +77,14 @@ spec:
                   type: string
                 defaultAdminUser:
                   type: string
+                osAuth:
```
Comment (ericabr) [inline deploy/olm-catalog/ibm-commonui-operator/1.1.0/operators.ibm.com_legacyheaders_crd.yaml]: remove this for legacy header

**Pull request merged**